File: Benchmarks/normal/java/spring/testcode/BenchmarkTest14704.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest14704 {

    private static String trimEnds(String v) { return v.trim(); }
    private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(BenchmarkTest14704.class);

    @GetMapping("/BenchmarkTest14704")
    public void BenchmarkTest14704(HttpServletRequest request, HttpServletResponse response) throws Exception {
        String propValue = java.util.Optional.ofNullable(System.getProperty("app.value", "")).orElse("");
        String data = trimEnds(propValue);
        LOG.info("Action: {}", data);
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}

File: Benchmarks/enterprise/python/flask/testcode/benchmark_test_14704.py
# SPDX-License-Identifier: Apache-2.0
from flask import request


def BenchmarkTest14704():
    user_id = request.args.get('id', '')
    data = user_id if user_id else 'default'
    with open('/var/app/data/' + str(data), 'r') as fh:
        content = fh.read()
    return content
